packages/core/src/models/MCP.ts

```diff
@@ -0,0 +1,244 @@
+/**
+ * @fileoverview MCP class for Composio SDK, used to manage MCP servers.
+ *
+ * @author Apoorv Taneja <apoorv@composio.dev>
+ * @date 2025-06-12
+ * @module MCP
+ */
+import ComposioClient from '@composio/client';
+import { telemetry } from '../telemetry/Telemetry';
+import {
+  MCPGenerateURLParams,
+  MCPCreateMethodResponse,
+  MCPToolkitConfig,
+  MCPAuthOptions,
+  MCPGetServerParams
+} from '../types/mcp.types';
+import {
+  McpCreateResponse,
+  McpListResponse,
+  McpUpdateResponse,
+  McpDeleteResponse,
+  CustomCreateResponse,
+  GenerateURLResponse
+} from '@composio/client/resources/mcp';
+
+/**
+ * MCP (Model Control Protocol) class
+ * Handles MCP server operations
+ */
+export class MCP {
+  private client: ComposioClient;
+
+  constructor(client: ComposioClient) {
+    this.client = client;
+    telemetry.instrument(this);
+  }
+
+  /**
+   * Create a new MCP server
+   * @param {string} name - Unique name for the MCP server
+   * @param {MCPToolkitConfig[]} toolkitConfigs - Array of toolkit configurations
+   * @param {MCPAuthOptions} [authOptions] - Authentication configuration options
+   * @returns {Promise<MCPCreateMethodResponse>} Created server details with instance getter
+   * 
+   * @example
+   * ```typescript
+   * const cfg = await composio.mcp.create(
+   *   "personal-mcp-server",
+   *   [
+   *     {
+   *       toolkit: "GMAIL",
+   *       authConfigId: "ac_sdhkjfhjksdk",
+   *       allowedTools: ["GMAIL_FETCH_EMAILS"],
+   *     },
+   *   ],
+   *   { useManagedAuthByComposio: true },
+   * );
+   * 
+   * // Get server instance
+   * const server = await cfg.getServer({
+   *   connected_account_ids: {
+   *     "gmail": "acc_123",
+   *     "supabase": "acc_456"
+   *   }
+   * });
+   * // OR
+   * const server = await cfg.getServer({
+   *   user_id: "user_123"
+   * });
+   * ```
+   */
+  async create(
+    name: string,
+    toolkitConfigs: MCPToolkitConfig[],
+    authOptions?: MCPAuthOptions
+  ): Promise<MCPCreateMethodResponse> {
+    // Check for unique toolkits
+    const toolkits = toolkitConfigs.map(config => config.toolkit);
+    const uniqueToolkits = new Set(toolkits);
+    if (uniqueToolkits.size !== toolkits.length) {
+      throw new Error('Duplicate toolkits are not allowed. Each toolkit must be unique.');
+    }
+
+    // Create a custom MCP server with the toolkit configurations
+    const mcpServerCreatedResponse: CustomCreateResponse = await this.client.mcp.custom.create({
```

Comment:
Add error handling here, and define and throw custom error in /errors

Reply:
added it.